src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java

```diff
@@ -162,17 +213,29 @@ private void cleanupWiremock(ExtensionContext context) {
                 if (server.isRunning()) {
                   server.resetRequests();
                   server.resetToDefaultMappings();
-                  server.stop();
                 }
               });
       instancesMap.clear();
     }
   }
 
-  private WiremockMap getWiremockMap(Store store) {
+  private void shutdownWiremock() {
+    wiremockMapLock.lock();
+
+    wiremockMap.values().forEach(WireMock::shutdown);
+    wiremockMap.clear();
+
+    wiremockMapLock.unlock();
+  }
+
+  private WiremockServerMap getWiremockServerMap(Store store) {
     return store.getOrComputeIfAbsent(
-        WIREMOCK_INSTANCES_MAP_STORE_KEY, o -> new WiremockMap(), WiremockMap.class);
+        WIREMOCK_SERVER_INSTANCES_MAP_STORE_KEY,
+        o -> new WiremockServerMap(),
+        WiremockServerMap.class);
   }
 
-  private class WiremockMap extends HashMap<Integer, WireMockServer> {}
+  private static class WiremockMap extends HashMap<Integer, WireMock> {}
```

Comment:
Wouldn't be simpler to just have `wiremockMap`  defined as `HashMap<Integer, WireMock>`?